internal/transport/http_util.go

```diff
@@ -332,19 +334,33 @@ func (w *bufWriter) Write(b []byte) (n int, err error) {
 		n, err = w.conn.Write(b)
 		return n, toIOError(err)
 	}
+	if w.buf == nil {
+		b := w.pool.Get().(*[]byte)
+		w.buf = *b
+	}
 	for len(b) > 0 {
-		nn := copy(w.buf[w.offset:], b)
+		nn := copy((w.buf)[w.offset:], b)
```

Comment:
Do we need the parentheses around `w.buf`?

Reply:
fixed